specs/_features/maxeb_increase/capella.py

```diff
@@ -3721,54 +3685,39 @@ def get_expected_withdrawals(state: BeaconState) -> Sequence[Withdrawal]:
     validator_index = state.next_withdrawal_validator_index
     withdrawals: List[Withdrawal] = []
     consumed = 0
-    for withdrawal in state.eligible_slashing_withdrawals:
-        if withdrawal.withdrawable_epoch > epoch or consumed == MAX_WITHDRAWALS_PER_PAYLOAD:
+    for withdrawal in state.pending_partial_withdrawals:
+        if withdrawal.withdrawable_epoch > epoch or len(withdrawals) == MAX_WITHDRAWALS_PER_PAYLOAD // 2:
             break
 
         validator = state.validators[withdrawal.index]
-        withdrawable_balance = state.balances[validator.index]
-        if withdrawable_balance > Gwei(0):
+        if validator.exit_epoch == FAR_FUTURE_EPOCH and state.balances[withdrawal.index] > MIN_ACTIVATION_BALANCE:
+            withdrawable_balance = min(state.balances[withdrawal.index] - MIN_ACTIVATION_BALANCE, withdrawal.amount)
             withdrawals.append(Withdrawal(
                 index=withdrawal_index,
-                validator_index=validator_index,
+                validator_index=withdrawal.index,
                 address=ExecutionAddress(validator.withdrawal_credentials[12:]),
                 amount=withdrawable_balance,
             ))
             withdrawal_index += WithdrawalIndex(1)
-        consumed += 1
-    for withdrawal in state.eligible_balance_withdrawals:
-        if withdrawal.withdrawable_epoch > epoch or consumed == MAX_WITHDRAWALS_PER_PAYLOAD:
-            break
 
-        validator = state.validators[withdrawal.index]
+        consumed += 1
 
-        # Compute withdrawing balance
-        withdrawable_balance = Gwei(0)
-        if withdrawal.is_exit:
-            withdrawable_balance = state.balances[validator.index]
-        elif state.balances[validator.index] > MIN_ACTIVATION_BALANCE:
-            withdrawable_balance = min(state.balances[validator.index] - MIN_ACTIVATION_BALANCE, withdrawal.amount)
+    state.pending_partial_withdrawals = state.pending_partial_withdrawals[consumed:] 
 
-        if withdrawable_balance > Gwei(0):
+    # Sweep for remaining.
+    bound = min(len(state.validators), MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP)
+    for _ in range(bound):
+        validator = state.validators[validator_index]
+        balance = state.balances[validator_index]
+        if is_fully_withdrawable_validator(validator, balance, epoch):
```

Comment:
so the exits are back with the sweep partials?

Reply:
Yes, exactly